.github/workflows/tests.yml

```diff
@@ -24,3 +27,61 @@ jobs:
           && !contains(github.event.pull_request.labels.*.name, 'ci:run-full-test-suite')
           || false
          }}
+
+  upload-stable-deps:
+    needs: tests
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-tags: true
+          sparse-checkout: .github
+
+      - name: Prepare local repo
+        run: |
+          git fetch
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
+            git checkout "${{ env.DEPS_BRANCH }}"
+            git rebase origin/master
+          else
+            git checkout master
+            git checkout -b "${{ env.DEPS_BRANCH }}"
+          fi
+          rm -f .github/stable/*.txt
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: frozen-*
+          path: .github/stable/
+          merge-multiple: true
+
+      - name: Determine if changes have been made
+        id: changed
+        run: |
+          echo "has_changes=$(git status --porcelain | wc -l | awk '{print $1}')" >> $GITHUB_OUTPUT
+
+      - name: Stage changes
+        if: steps.changed.outputs.has_changes != '0'
+        run: |
+          git add .github/stable/
+          git commit -m "Update changed dependencies"
+          git push -f --set-upstream origin "${{ env.DEPS_BRANCH }}"
+
+      # Create PR to master
+      - name: Create pull request
+        if: steps.changed.outputs.has_changes != '0'
+        uses: repo-sync/pull-request@v2
+        with:
+          source_branch: "${{ env.DEPS_BRANCH }}"
+          destination_branch: "master"
+          github_token: "${{ secrets.GITHUB_TOKEN }}"
+          pr_title: "Update stable dependency files"
+          pr_body: "Automatic update of stable requirement files to snapshot valid python environments. Please push an empty commit using `git commit --allow-empty -m 'trigger ci'` as bots are not able to trigger CI on their own."

```

Comment:
# Suggestion
Could be worth moving the inline code block to a fenced code block so that it can be copied from the GItHub UI by clicking on it.

### Compare:

Hello, please `run this code`.

---

Hello, please

```
run this code
```

Reply:
done! see #5184 as an example

Reply:
Looks great! 🎉